cabra/src/pages/components/TaskRelationsEdit.tsx

```diff
@@ -0,0 +1,102 @@
+import tw, { styled } from "twin.macro";
+
+import { DirectedRelationType } from "../../types/taskRelation";
+import { ITask } from "../../types/task";
+import RelationListEdit from "./RelationListEdit";
+import Select from "react-select";
+import { getRelatedTasks } from "../../api/taskRelations";
+import { getTasks } from "../../api/tasks";
+import { useQuery } from "react-query";
+import { useState } from "react";
+
+const Container = styled.div(tw`text-gray-50 w-full space-y-4`);
+interface Props {
+  readonly taskId: string;
+}
+interface RelationshipOption {
+  readonly value: DirectedRelationType;
+  readonly label: string;
+}
+interface TaskOption {
+  value: ITask;
+  label: string;
+}
+
+const relationshipOptions: RelationshipOption[] = [
+  { value: DirectedRelationType.DependsOn, label: "Depends on" },
+  { value: DirectedRelationType.IsDependentOn, label: "Is dependent on" },
+  { value: DirectedRelationType.Interchangable, label: "Interchangeable" },
+];
+
+const DEFAULT_RELATION_TYPE = DirectedRelationType.DependsOn;
+
+export default function TaskRelationsEdit({ taskId }: Props) {
+  const [tasksRelation, setTaskRelation] = useState<DirectedRelationType>(
+    DEFAULT_RELATION_TYPE
+  );
+  const [relatedTasks, setRelatedTasks] = useState<ITask[]>([]);
+  const [showSelected, setShowSelected] = useState(false);
+
+  const { data: dataSubtasks } = useQuery(
+    getRelatedTasks.cacheKey(taskId, DirectedRelationType.Subtask),
+    () => getRelatedTasks.run(taskId, DirectedRelationType.Subtask)
+  );
+  const { data, isLoading, error } = useQuery(getTasks.cacheKey, getTasks.run);
+
+  if (isLoading) return <Container>Loading</Container>;
+  if (error || !data?.data) return <Container>Oops</Container>;
+
+  const subtasks = dataSubtasks?.data;
+
+  const potentialRelative = (isDone: boolean, id: string) => {
+    return (
+      !isDone &&
+      id !== taskId &&
+      !subtasks?.map(({ task: subtask }) => subtask.id).includes(id)
+    );
+  };
+
+  const formattedRelativesToBe: TaskOption[] = data.data
+    .filter(({ isDone, id }) => potentialRelative(isDone, id))
+    .map((task) => ({ value: task, label: task.title }));
+
+  const toggleShowSelected = () => {
+    setShowSelected(!showSelected);
+  };
+
+  return (
+    <>
```

Comment:
Thats an unnecessary empty markup, should be replaced with Container or div.

Reply:
That's not empty markup, it's a `React.Fragment` and it has to be there because components must return [JSX with a single root element](https://beta.reactjs.org/learn/writing-markup-with-jsx#1-return-a-single-root-element).